tests/s25Main/UI/testVideoDriver.cpp

```diff
@@ -39,3 +43,58 @@ BOOST_AUTO_TEST_CASE(FindClosestVideoMode)
     BOOST_TEST(driver->FindClosestVideoMode(VideoMode(2000, 622)) == VideoMode(1500, 800));
     BOOST_TEST(driver->FindClosestVideoMode(VideoMode(1200, 900)) == VideoMode(1500, 800));
 }
+
+namespace rttrOglMock2 {
+RTTR_IGNORE_DIAGNOSTIC("-Wmissing-declarations")
+
+std::vector<GLuint> activeTextures;
+
+void APIENTRY glGenTextures(GLsizei n, GLuint* textures)
+{
+    static GLuint cur = 0;
+    for(; n > 0; --n)
+    {
+        *(textures++) = ++cur;
+        activeTextures.push_back(cur);
+    }
+}
+void APIENTRY glDeleteTextures(GLsizei n, const GLuint* textures)
+{
+    for(; n > 0; --n)
+    {
+        BOOST_TEST(*textures != 0u);
+        BOOST_TEST(helpers::contains(activeTextures, *textures));
+        helpers::remove(activeTextures, *(textures++));
+    }
+}
+
+RTTR_POP_DIAGNOSTIC
+} // namespace rttrOglMock2
+
+BOOST_AUTO_TEST_CASE(CreateAndDestroyTextures)
+{
+    // Fresh start
+    VIDEODRIVER.DestroyScreen();
+    VIDEODRIVER.CreateScreen(VideoMode(800, 600), false);
+
+    glGenTextures = rttrOglMock2::glad_glGenTextures;
+    glDeleteTextures = rttrOglMock2::glDeleteTextures;
+    for(unsigned i = 1u; i <= 5u; ++i)
+        BOOST_TEST(VIDEODRIVER.GenerateTexture() == i);
+    BOOST_TEST_REQUIRE(rttrOglMock2::activeTextures.size() == 5u);
+    VIDEODRIVER.DestroyScreen();
+    BOOST_TEST_REQUIRE(rttrOglMock2::activeTextures.empty());
+    // Next cleanup call is a no-op (validated inside glDeleteTextures)
+    VIDEODRIVER.CleanUp();
+
+    VIDEODRIVER.CreateScreen(VideoMode(800, 600), false);
+    glGenTextures = rttrOglMock2::glad_glGenTextures;
```

Comment:
isn't it called glGenTextures and not glad_glGenTextures ?

Reply:
true. My IDE was smart enough to look through the macro ^^